src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
+{
+    my ($self) = @_;
+    my $is_newer = 0;   # Assume false
+    if (  exists(*$self->{options}->{source}) && $self->_is_valid_source(*$self->{options}->{source}) ) {
+        # stat()[9] is modification time
+        if ( !$self->_is_valid_source(*$self->{filename}) || 
+             ((stat(*$self->{options}->{source}))[9] > (stat(*$self->{filename}))[9]) ) {
+          $is_newer = 1
+        }
+    }
+    #FIXME: replace by $self->debug() after PR #154 has been merged...
+    if ( *$self->{LOG} ) {
+        if ( $is_newer ) {
+            # As this is a non reproducible event, be sure to log it when it happens
+            *$self->{LOG}->info("File ", *$self->{filename}, " contents reset to reference file (", *$self->{options}->{source}, ") contents.");
+        } else {
+            *$self->{LOG}->debug(1, "Reference file (", *$self->{options}->{source}, ") older than ", *$self->{filename}, ": ignoring it");
+        };
+    };
+    return $is_newer;
+}
 
 sub new
 {
     my $class = shift;
     my $self = $class->SUPER::new (@_);
-    if ($self->_is_valid_source(*$self->{filename})) {
-        my $txt = LC::File::file_contents (*$self->{filename});
+    my $src_file;
+    my ($path, %opts) = @_;
+
+    *$self->{options}->{source} = $opts{source} if exists ($opts{source});
+    if ( $self->_is_reference_newer() ) {
+        $src_file = *$self->{options}->{source};
+    } elsif ($self->_is_valid_source(*$self->{filename})) {
```

Comment:
@jouvin heh, that is indeed the case. i was unaware of this. i opened https://github.com/quattor/CAF/issues/157, but i already do not see how this could be handled cleanly or what it would add :confused: